Tests/Kernel/TestSendMsg.cpp

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2023, Agustin Gianni <agustingianni@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/NumericLimits.h>
+#include <LibCore/System.h>
+#include <LibTest/TestCase.h>
+#include <arpa/inet.h>
+#include <errno.h>
+
+static ErrorOr<ssize_t> sendmsg_helper(const struct msghdr* msg)
+{
+    auto fd = TRY(Core::System::socket(AF_INET, SOCK_DGRAM, 0));
+
+    sockaddr_in addr = {};
+    addr.sin_family = AF_INET;
+    addr.sin_port = htons(3333);
+    addr.sin_addr.s_addr = INADDR_ANY;
+
+    auto connectError = Core::System::connect(fd, (struct sockaddr const*)&addr, sizeof(addr));
```

Comment:
I'm less of a stickler for names being "long and descriptive" (e.g. `ret` or `err`) than other maintainers, but we should at least stick to `snake_case` variable names, even in tests